### PR TITLE
Completed react-hot-button component exercise

### PR DESCRIPTION
**Pull request by wang-morris**
https://user-images.githubusercontent.com/103090771/208269362-b31e32e6-c8e5-4f6a-b1cf-1f3c9b2fcdfd.mp4

### CODE REVIEW

**Comment by wang-morris** [comment]
Did not use both .jsx files in the first commit.

File: react-hot-button-component/src/hot-button.jsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+let clickCount = 0;
+
+class CustomButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { isClicked: false };
```
Comment (scbowler) [inline react-hot-button-component/src/hot-button.jsx]: The click count should be stored in state:

```JAVASCRIPT
this.state = {
  count: 0
};
```

File: react-hot-button-component/src/hot-button.jsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  }
+
+  handleClick(props) {
+    this.setState({ isClicked: true });
+  }
```
Comment (scbowler) [inline react-hot-button-component/src/hot-button.jsx]: Here the count should be incremented by 1:
```JAVASCRIPT
handleClick() {
  this.setState({ count: this.state.count + 1 });
}
```

File: react-hot-button-component/src/hot-button.jsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        </div>
+      );
+    } else {
+      return (
+        <div>
+          <button onClick={this.handleClick.bind(this)} className="super-hot">Hot Button</button>
+        </div>
+      );
+    }
+  }
```
Comment (scbowler) [inline react-hot-button-component/src/hot-button.jsx]: In `render` you can use `this.state.count` to determine which class should be applied to the button, then return a single button at the end.
```JAVASCRIPT
render() {
  const count = this.state.count;
  let btnClass;

  if (count < 4) {
    btnClass = 'cold';
  } else if (count < 7) {
    btnClass = 'cool';
  } else if (count < 10) {
    btnClass = 'luke-warm';
  } else if (count < 13) {
    btnClass = 'warm';
  } else if (count < 16) {
    btnClass = 'hot';
  } else {
    btnClass = 'super-hot';
  }

  return (
    <div>
      <button onClick={this.handleClick.bind(this)} className={btnClass}>Hot Button</button>
    </div>
  );
}
```

**Review by scbowler — CHANGES_REQUESTED** [review]
Good job getting this to work, but here are some things to refactor. Let me know if you have any questions.

**Review by scbowler — APPROVED** [review]
Great job implementing the feedback, looks good! 👍

**Pull request merged**